### PR TITLE
fix(AppResources): try recover if resources update fails

### PR DESCRIPTION
**Pull request by KristianDD**
Fixes https://github.com/NativeScript/nativescript-cli/issues/3510

### CODE REVIEW

**Comment by KristianDD** [comment]
@rosen-vladimirov @Pip3r4o

File: lib/services/android-resources-migration-service.ts [hunk 1]
```diff
@@ -22,6 +22,22 @@ export class AndroidResourcesMigrationService implements IAndroidResourcesMigrat
...
+
+		try {
+			await this.tryMigrate(originalAppResources, appResourcesDestination, appResourcesBackup);
+			this.$logger.out(`Successfully updated your project's application resources '/Android' directory structure.${EOL}The previous version of your Android application resources has been renamed to '/${AndroidResourcesMigrationService.ANDROID_DIR_OLD}'`);
+		} catch (error) {
+			try {
+				this.recover(originalAppResources, appResourcesDestination, appResourcesBackup);
+				this.$logger.out("Failed to update resources. They should be in their initial state.");
+			} catch (error) {
+				this.$logger.out(`Failed to update resources.${EOL} Backup of original content is inside "${appResourcesBackup}".${EOL}If "${originalAppResources} if missing copy from backup folder."`);
```
Comment (rosen-vladimirov) [inline lib/services/android-resources-migration-service.ts]: `If "${originalAppResources} iS missing...`

Comment (rosen-vladimirov) [reply lib/services/android-resources-migration-service.ts]: Why don't we throw the original error here? Currently the command exits with code 0, as we've caught all errors, which is incorrect.

Comment (rosen-vladimirov) [inline lib/services/android-resources-migration-service.ts]: Can you also add `this.$logger.trace` with the currently caught error?

**Pull request merged**